pandas/core/groupby/generic.py

```diff
@@ -230,6 +227,18 @@ def apply(self, func, *args, **kwargs):
     )
     def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs):
 
+        if maybe_use_numba(engine):
```

Comment:
would not object to making a _aggregate_with_python_cython (where you put everything L242 and on down.

Reply:
I could do this in a follow up refactor PR.

I guess I would need to make a Series and DataFrame version of this function since it looks like both are different.